### DISCUSSION TITLE
Cors implementation

### DISCUSSION THREAD

**Discussion by jigmeloday** [jigmeloday/blog-post-api#13] [Q&A] [answered]
How to implement Cors in rails?

**Comment by cybertronbaka** [comment, marked as answer]
To implement CORs is Rails, you will need a Gem called `rack-cors`.

1. Install the gem
`gem 'rack-cors'`
2. create a file ` config/initializers/cors.rb`
3. In the file, add the following:
```ruby
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'localhost'

    resource '*',
             headers: :any,
             methods: %i[get post put patch delete options head],
             credentials: true
  end
end
```

Please change the `origins`, `resource`, `headers`, `methods` and `credentials` to what you need.

**Reply by jigmeloday** [reply]
Thank you
